packages/flutter/test/foundation/service_extensions_test.dart

```diff
@@ -104,7 +104,18 @@ void main() {
   test('Service extensions - pretest', () async {
     binding = TestServiceExtensionsBinding();
     expect(binding.frameScheduled, isTrue);
+
+    // We need to test this service extension here because the result should
+    // always be true after the first binding.doFrame() call.
+    Map<String, dynamic> firstFrameResult;
+    firstFrameResult = await binding.testExtension('didSendFirstFrameEvent', <String, String>{});
```

Comment:
add a check that
expect(binding.debugDidSentFirstFrame, isFalse);
and
expect(binding.debugDidSentFirstFrame, isTrue);

placing those tests before the call to the extension.  That should self document what the expected behavior is and make any bugs easier to track down.

Reply:
Done.